nemo/collections/asr/modules/rnn_encoder.py

```diff
@@ -0,0 +1,187 @@
+        bidirectional (float): specifies whether RNN layers should be bidirectional or not
+            Defaults to True.
+        feat_out (int): the size of the output features
+            Defaults to -1 (means feat_out is d_model)
+        subsampling (str): the method of subsampling, choices=['stacking, 'vggnet', 'striding']
+            Defaults to stacking.
+        subsampling_factor (int): the subsampling factor
+            Defaults to 4.
+        subsampling_conv_channels (int): the size of the convolutions in the subsampling module for vggnet and striding
+            Defaults to -1 which would set it to d_model.
+        dropout (float): the dropout rate used between all layers
+            Defaults to 0.2.
+    """
+
+    def input_example(self):
+        """
+        Generates input examples for tracing etc.
+        Returns:
+            A tuple of input examples.
+        """
+        input_example = torch.randn(16, self._feat_in, 256).to(next(self.parameters()).device)
+        input_example_length = torch.randint(0, 256, (16,)).to(next(self.parameters()).device)
+        return tuple([input_example, input_example_length])
+
+    @property
+    def input_types(self):
+        """Returns definitions of module input ports.
+        """
+        return OrderedDict(
+            {
+                "audio_signal": NeuralType(('B', 'D', 'T'), SpectrogramType()),
+                "length": NeuralType(tuple('B'), LengthsType()),
+            }
+        )
+
+    @property
+    def output_types(self):
+        """Returns definitions of module output ports.
+        """
+        return OrderedDict(
+            {
+                "outputs": NeuralType(('B', 'D', 'T'), AcousticEncodedRepresentation()),
+                "encoded_lengths": NeuralType(tuple('B'), LengthsType()),
+            }
+        )
+
+    def __init__(
+        self,
+        feat_in,
+        n_layers,
+        d_model,
+        proj_size=-1,
+        rnn_type='lstm',
+        bidirectional=True,
+        subsampling='striding',
+        subsampling_factor=4,
+        subsampling_conv_channels=-1,
+        dropout=0.1,
+    ):
+        super().__init__()
+
+        self.d_model = d_model
+        self._feat_in = feat_in
+
+        if subsampling_conv_channels == -1:
+            subsampling_conv_channels = proj_size
+        if subsampling and subsampling_factor > 1:
+            if subsampling == 'stacking':
+                self.pre_encode = StackingSubsampling(
+                    subsampling_factor=subsampling_factor, feat_in=feat_in, feat_out=proj_size
+                )
+            else:
+                self.pre_encode = ConvSubsampling(
+                    subsampling=subsampling,
+                    subsampling_factor=subsampling_factor,
+                    feat_in=feat_in,
+                    feat_out=proj_size,
+                    conv_channels=subsampling_conv_channels,
+                    activation=nn.ReLU(),
```

Comment:
Not swish? Should this be user-modifiable ?

Reply:
Lets experiment first, we can make it user configurable via config later if it helps.